utils/vim/syntax/swift.vim

```diff
@@ -140,7 +140,7 @@ syn match swiftTypeDeclaration skipwhite nextgroup=swiftType
       \ /->/
 
 syn region swiftString start=/"/ skip=/\\\\\|\\"/ end=/"/ contains=swiftInterpolation
-syn region swiftInterpolation start=/\\(/ end=/)/ contained
+syn region swiftInterpolation start=/\\(/ end=/)/ contained contains=swiftPreproc
```

Comment:
Good point. Really, I'd prefer if nothing but the delimiters were highlighted - this seems to fit more closely with how Xcode is doing it. IMO the `foo` in `"\(foo)"` shouldn't be highlighted. But I don't exactly know how to do that. I think I got close with:

```vim
syn region swiftStringFull start=/"/ skip=/\\\\\|\\"/ end=/"/ contains=TOP,swiftString keepend transparent
syn match swiftString /[")][^"]\{-}\("\|\\(\)/ contained
```

I could certainly add those other categories and keep the existing relationship between `swiftString` and `swiftInterpolation` if you think that's easiest.